Library/Homebrew/formula_installer.rb

```diff
@@ -344,7 +344,8 @@ def expand_dependencies(deps)
     inherited_options = {}
 
     expanded_deps = Dependency.expand(formula, deps) do |dependent, dep|
-      options = inherited_options[dep.name] = inherited_options_for(dep)
+      inherited_options[dep.name] ||= Options.new
+      options = inherited_options[dep.name] += inherited_options_for(dep)
```

Comment:
I found the previous double assignment not exactly readable and it only got worse. How about the following?

```ruby
inherited_options[dep.name] ||= Options.new
inherited_options[dep.name] += inherited_options_for(dep)
options = inherited_options[dep.name]
```

And maybe just drop the assignment to `options` completely? If I'm not overlooking something, it is only used once in the following code and that instance could as well be replaced with `inherited_options[dep.name]`.